windows/src/desktop/kmshell/install/UfrmInstallKeyboardFromWeb.pas

```diff
@@ -117,7 +118,17 @@ procedure TfrmInstallKeyboardFromWeb.TntFormCreate(Sender: TObject);
 
 procedure TfrmInstallKeyboardFromWeb.cefBeforeBrowseSync(Sender: TObject; const Url: string; out Handled: Boolean);
 begin
-  Handled := TRegEx.IsMatch(Url, UrlPath_RegEx_MatchKeyboardsInstall);
+  // This introduces some deeper knowledge of URL paths in Keyman Configuration, which is
+  // a bit of a shame, because we try to keep our internal knowledge to /go/ urls on
+  // keyman.com. However, there is not really any great way around this that I've found,
+  // which allows us to handle internal navigation on the keyboard search and still lets
+  // us open other URLs that may be in the search results in an external browser.
+  Handled :=
+    TRegEx.IsMatch(Url, URLPath_RegEx_MatchKeyboardsInstall) or       // capture https://keyman.com/keyboards/install/*
+    (not TRegEx.IsMatch(Url, UrlPath_RegEx_MatchKeyboardsRoot) and    // don't capture https://keyman.com/keyboards*
+    not Url.StartsWith('keyman:') and                                 // don't capture keyman:*
+    not IsLocalUrl(Url) and                                           // don't capture the external frame or its resources
+    not TRegEx.IsMatch(Url, UrlPath_RegEx_MatchKeyboardsGo));         // don't capture the launch url https://keyman.com/go/windows/download-keyboards
```

Comment:
@jahorton @darcywong00 @ermshiperete this pattern here is similar to what you'll need in your platform implementations for determining if a URL in the embedded download dialog is internal or external.

Reply:
... uh-huh.  For what purpose(s)?  To block any external links?

Reply:
Is the plan to open any external links (from keyboard search) outside the app?  So we vet any links opened during keyboard search and kick the user out to a browser if the link isn't also part of the keyboard search?

Reply:
> So we vet any links opened during keyboard search and kick the user out to a browser if the link isn't also part of the keyboard search?

Correct.

Reply:
Although ... that pattern makes heaps of sense on desktop. I'm not so sure it works as well on mobile devices. Something to think about.

Reply:
It's... not a bad paradigm for iOS, at least recent versions.  Whenever the user wishes to return to the app, there's a "Back to Keyman" button at the top right.